### PR TITLE
Don't double up on periods in anomalies

### PR DESCRIPTION
**Pull request by JasonGross**
We don't want "Anomaly: Returned a functional value in a type not
recognized as a product type.. Please report at
http://coq.inria.fr/bugs/." but instead "Anomaly: Returned a functional
value in a type not recognized as a product type. Please report at
http://coq.inria.fr/bugs/."

### CODE REVIEW

**Comment by JasonGross** [comment]
@maximedenes Please put this in the 8.7 milestone

**Comment by herbelin** [comment]
Shouldn't it be the "." in the `Anomaly` line of `lib/cErrors.ml` which should be removed? How to do otherwise with messages of the form: "id should already exist!"? Then a period should be added for consistency in other messages, but there are more, that's true.

**Comment by JasonGross** [comment]
I can do that instead, if that's preferred?  I have no strong opinion, but don't want to bounce back and forth.  I'll plan to make a commit doing that later today, if there's no objection to it.  (I'm also curious if @ejgallego agrees.)

**Comment by ejgallego** [comment]
No strong opinion on that, these issues are subtle, we could try to inspect the pp document but that is not easy.

For 8.8 I'd like to propose some minor changes to printing, there are still some minor rough edges resulting from the transition from the old console-based printing to the "feedback" based one, and I am not happy with the status quo. In particular we still have two systems to handle printing priorities (`Flags.quiet` vs `Feedback.level`) and also the combinators in `Feedback` may not be enough for all cases.

**Comment by JasonGross** [comment]
@herbelin I've swapped where the `.` is inserted.

**Comment by herbelin** [comment]
Great. Thanks very much.

**Comment by maximedenes** [comment]
Doesn't seem to compile.

File: library/heads.ml [hunk 1]
```diff
@@ -72,7 +72,7 @@ let kind_of_head env t =
        with Not_found ->
          CErrors.anomaly
            Pp.(str "constant not found in kind_of_head: " ++
-               str (Names.Constant.to_string cst)))
+               str (Names.Constant.to_string cst ++ str ".")))
```
Comment (gares) [inline library/heads.ml]: This should be `^` not `++ str`

Comment (ejgallego) [inline library/heads.ml]: Even better use `Constant.print cst ++ ...`

Comment (JasonGross) [inline library/heads.ml]: Thanks, fixed

Comment (ejgallego) [inline library/heads.ml]: Cool

**Comment by JasonGross** [comment]
Successful travis run: https://travis-ci.org/JasonGross/coq/builds/238961462

**Pull request merged**